code/game/objects/effects/chem/chemsmoke.dm

```diff
@@ -162,6 +162,17 @@
 					continue
 				else if (ismob(A))
 					chemholder.reagents.touch_mob(A)
+					if(istype(A, /mob/living/carbon))
+						var/mob/living/carbon/H = A
+						var/internals = H.get_breath_from_internal()
+						var/gasmask = FALSE
+						if(H.wear_mask)
+							gasmask = (H.wear_mask.item_flags & BLOCK_GAS_SMOKE_EFFECT)
+						to_chat(world, "gasmaks: [gasmask]")	
+						to_chat(world, "internals: [internals]")					
+						if(!internals && !gasmask)
+							chemholder.reagents.trans_to_mob(H, 5, CHEM_INGEST, copy = 1)
+							chemholder.reagents.trans_to_mob(H, 5, CHEM_BLOOD, copy = 1)
```

Comment:
DGotta remove debug messages , no need for () in the & check , and why are you using copy = 1 , and doing a double stack on the chems , this is both unnecesarry and will let people dupe reagents. Most predominanty , since you put copy to true , the cloud will never run out of reagents , which combined with the double transfer , will let people dupe reagents extremly easily (can't wait to see moebius gamers doing 300u uranium 1 tile nades and then having 90000 uranium reagent in their body) Should just transfer to blood a fixed amount and not let it copy..

Reply:
Forgot about the messages. I just copied the rest from the chemsmoke cloud where it worked the same.